### DISCUSSION TITLE
Running workflows without parallel jobs

### DISCUSSION THREAD

**Discussion by cinnion** [nektos/act#2512] [Q&A]
In looking at the output from act, it appears that jobs are being run in parallel. Is this so? And if so, is there a way to turn off running them in parallel and running them one after another. I am getting collisions on docker container names as a result of this apparent parallelism.

**Comment by ChristopherHX** [comment]
Yes the workflow is run parallel as much cpu cores you have (except the weird algorithm that throttles parallel execution by planning stages)
- dependent jobs run one after the other
- max-parallel throttles matrix job entries
- use `-W`, `-j` and `--matrix` flags to run a single workflow job

> is there a way to turn off running them in parallel

No such thing has ever been implemented in act, but yes editing the parallel executor file (replace ncpu by 1) https://github.com/nektos/act/blob/0de940b948e0397c2db8f98b82dd20d88a542325/pkg/runner/runner.go#L204 can let you change this. Then just install the golang sdk and run `go build` in the source code of act to get your binary.

**Reply by cinnion** [reply]
Thanks @ChristopherHX.  It may be awhile, but when I get a few cycles free, I will see if I can add another flag (say `-max-cpus` or even just `-ncpu`) to set the value for ncpu from the command line without someone having to edit it to change it. In the meantime, here is the solution I came up with using `-j` before I read the response.

[A shell script to run all the workflow jobs one by one](https://gist.github.com/cinnion/d55978bbc26f88fe7aaba833d3e5787d)

**Reply by ChristopherHX** [reply]
One problem for those using `needs.<job>.outputs` etc this wouldn't work. 

> few cycles free

Yes this is always a problem, for me is to get reviews for enhancements as wells. I'm throtteling to limit my open PR's to be manageable.

It would be so nice to have more people as members who can decide about merging a PR, even if those don't write code.
